File: backend/src/main/java/com/hgw/gestionale/user/controller/UserController.java
package com.hgw.gestionale.user.controller;

import com.hgw.gestionale.user.dto.UpdateUserRequest;
import com.hgw.gestionale.user.dto.UserResponse;
import com.hgw.gestionale.user.service.UserService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.security.core.Authentication;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping(value = "/api/users", produces = MediaType.APPLICATION_JSON_VALUE)
@RequiredArgsConstructor
public class UserController {
    private final UserService userService;

    @GetMapping("/me")
    public UserResponse getCurrentUser(Authentication authentication) {
        return userService.getCurrentUser(authentication.getName());
    }

    @PutMapping("/me")
    public UserResponse updateCurrentUser(
            @Valid @RequestBody UpdateUserRequest request,
            Authentication authentication
    ) {
        return userService.updateCurrentUser(authentication.getName(), request);
    }
}

File: backend-node/src/controller/userController.ts
import {
    Body,
    Controller,
    Get,
    OperationId,
    Put,
    Request,
    Route,
    Security,
    Tags,
} from "tsoa";
import { UserResponse, UpdateUserRequest } from "../dto/user.dto";
import { UserService } from "../service/user.service";
import { AccessTokenPayload } from "../common/token";
import express from "express";

const userService = new UserService();

@Route("users")
@Tags("Users")
@Security("bearerAuth")
export class UserController extends Controller {
    @Get("/me")
    @OperationId("getCurrentUser")
    public async getCurrentUser(@Request() request: express.Request): Promise<UserResponse> {
        const username = ((request as unknown as { user: AccessTokenPayload }).user).sub!;
        return userService.getByUsername(username);
    }

    @Put("/me")
    @OperationId("updateCurrentUser")
    public async updateCurrentUser(
        @Body() body: UpdateUserRequest,
        @Request() request: express.Request
    ): Promise<UserResponse> {
        const username = ((request as unknown as { user: AccessTokenPayload }).user).sub!;
        return userService.update(username, body);
    }
}
